internal/controllers/machine/machine_controller.go

```diff
@@ -488,6 +499,74 @@ func (r *Reconciler) isDeleteNodeAllowed(ctx context.Context, cluster *clusterv1
 	return nil
 }
 
+// isVersionAllowed returns true if the machine version conforms to the Kubernetes version skew policy.
+// If the Machine is part of the control plane, then its major.minor version must be greater than or equal to the
+// control plane actual major.minor version, and less than or equal to the control plane desired major.minor version.
```

Comment:
I could be wrong, but could it be that it's possible to downgrade with KCP today?
(I only found something that it's forbidden via ClusterClass)

If I'm correct we should probably start blocking downgrades in KCP? (separately from this PR)

Reply:
> If I'm correct we should probably start blocking downgrades in KCP? (separately from this PR)

Probably.

The current version skew policy does not mention downgrade. The most recent discussion took place in 2019-2020: https://github.com/kubernetes/website/issues/12327.